### ISSUE TITLE
Selecting multiple files in a open file dialog is buggy

### ISSUE THREAD

**Issue by wout** [elementary/files#1233]
## Prerequisites
- [X] I have searched open and closed issues for duplicates.

## Describe the bug
In Thunderbird, for example, the file browser to add attachments, does not allow to select more than two files using `ctrl` + `click`. The first file, works, the second one too, but from the third file onward, behaviour gets buggy. Files can be (de)selected by dragging a select box over them, but once they are deslected, the same behaviour is true. Selecting two files works, but the third, fourth, ... not.

This is not only true for Thunderbird, but for all apps using the elementary file browser.

Please have a look at the screencast to get an idea of what I mean:

[Screen record from 2020-02-11 11.56.45@2x.mp4.zip](https://github.com/elementary/files/files/4185587/Screen.record.from.2020-02-11.11.56.45%402x.mp4.zip)

This bug has been around for quite some time.

### To Reproduce
1. Open a file browser in Thunderbird using the attachemnts section in a new email.
2. Hold down `ctrl` and click on the files to be selected.
3. Try to select mor than two files.

### Expected behavior
Files should be selected while clicking on them.

## Platform Information
![image](https://user-images.githubusercontent.com/107324/74235853-d42ad080-4cc7-11ea-8e6a-60161db7f716.png)

- [ ] I'm using the latest version from git that I've manually compiled
- [X] I'm using the latest released stable version

**Comment by jeremypw** [comment]
The list box part of the filechooser is an unmodified part of Gtk.FileChooser - only the pathbar is customised so I am not sure this is a Files issues.  I tried selecting multiple files with Ctrl+Click using Code to open the dialog. This worked OK as long as you did not select a folder (since it was expecting documents). However I noticed that deselecting a file by clicking on it again did not always work first time.  Note I did not have the preview active.  I'll try again with FireFox and the preview panel.

**Comment by jeremypw** [comment]
How did you set the preview widget?

**Comment by wout** [comment]
I am not sure what you mean with the preview widget.

**Comment by jeremypw** [comment]
I noticed in your video that when you clicked on a jpg file in the chooser, a preview of it appeared on the right.  This is not standard chooser behaviour.

**Comment by jeremypw** [comment]
Yes - this is a FireFox "feature". WIth FireFox you seem to only be able to open one file at a time - I now notice you used Thunderbird in your demo which I do not have installed. However, using Chromium to attach files to a test email also gave a preview.  I too found that deselecting/reselecting files when several were selected was a bit unreliable - worse than with Code.  Sometimes had to click several times to get a reaction.  I suspect that overhead of the preview widget maybe at least partly to blame. 

I cannot confirm the bug as described - I had no trouble selecting multiple files.  It was deselecting and reselecting when there were several files selected that seems buggy.  Really need to test an app which used the native GtkFileChooser (from Bionic) for comparison as there is no obvious reason why Files should cause this.

**Comment by wout** [comment]
Ok, I suspected this file browser is not 100% elementary. But I am experiencing this issue everywhere I need to select multiple files. And I did not install any add-ons or plugins, jus standard elementary OS as much as possibble. :smile:

**Comment by jeremypw** [comment]
The intention is at some point to replace the customised Gtk.FileChooser with a 100% elementary FileChooser using the Gtk.NativeFileChooser portal. We can then ensure features and behaviour match Files and diagnose this kind of problem.  I'll try to test in a vanilla Ubuntu install without the Pantheon desktop.

**Comment by wout** [comment]
Great! Let me know if I can provide you with more information.

**Comment by jeremypw** [comment]
I did not get this issue at all when using FireFox on Ubuntu 18.04.4 LTS so it looks like it is something to do with elementary mods :-(  I'll see whether I can track it down.

**Comment by wout** [comment]
Have you checked Thunderbird with a multi-file select?

**Comment by wout** [comment]
I noticed something else that might be related. When the file dialog opens, the first file is selected. Holding down `ctrl`, I can deselect it and select it again, but after that, I am unable to deselect it again by clicking. Only dragging a select box, entering and leaving the file would deselect it. If that makes sense. :smile:

**Comment by jeremypw** [comment]
Added to project

**Comment by jeremypw** [comment]
Closing as this does not seem to be an issue with parts of the FileChooser under Files control. Moreover I did not experience this issue with a recent version of the FileChooser when using e.g. Code.
